src/main/java/javax/validation/valueextraction/ValueExtractor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Bean Validation API
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package javax.validation.valueextraction;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.validation.Path;
+
+/**
+ * Defines the logic used to extract the values to validate from an object of type {@code T}.
+ * <p>
+ * A {@link ValueExtractor} is tied to one specific type parameter of {@code T}. The {@link ExtractedValue} annotation
+ * is used to mark that type parameter.
+ * <p>
+ * The extracted values are passed to the corresponding method of the {@link ValueReceiver}.
```

Comment:
It'd be good to give an example definition showing the usage of `@ExtractedValue` in the JavaDocs.